saleor/graphql/product/tests/test_product.py

```diff
@@ -2427,6 +2430,15 @@ def test_search_product_by_description(user_api_client, product_list, channel_US
 
     assert len(content["data"]["products"]["edges"]) == 1
 
+    variables = {"filters": {"search": "red"}, "channel": channel_USD.slug}
```

Comment:
IMO these should be separate tests, responsible for testing the search when we need to take into consideration the weights of the fields. 
Also:
-  I would explicitly add the `red` word to the product description. Now, it is hard to understand, why the query returns two products. 
- I would use more objects for this test(remember about bulk_create). Product1 doesn't have `red` word, product 2 has `red` in the description, product 3 has two `red` words in the description, product 4 has `red` in name. The expected results should be - product4, product3, product2
- Also, DB returns products sorted by `slug`. I would also add different slugs to the products, to be fully sure that, returned products are sorted by search, and not by default slug sorting.